pkg/api/v1/control-plane/cache/simple.go

```diff
@@ -139,10 +139,15 @@ func (cache *snapshotCache) SetSnapshot(node string, snapshot Snapshot) error {
 				if cache.log != nil {
 					cache.log.Infof("respond open watch %d%v with new version %q", id, watch.Request.ResourceNames, version)
 				}
-				cache.respond(watch.Request, watch.Response, snapshot.GetResources(watch.Request.TypeUrl).Items, version)
 
-				// discard the watch
-				delete(info.watches, id)
+				resources := snapshot.GetResources(watch.Request.TypeUrl).Items
+				if resources != nil {
+					// snapshot has been initialized and exists
```

Comment:
let's expand the comment to include the different between nil and zero, as this is main motivation for this change